src/CourseAuthoringPage.tsx

```diff
@@ -1,40 +1,37 @@
 import React, { useEffect } from 'react';
-import PropTypes from 'prop-types';
 import { useDispatch, useSelector } from 'react-redux';
 
 import {
   useLocation,
 } from 'react-router-dom';
 import { StudioFooterSlot } from '@edx/frontend-component-footer';
 import Header from './header';
-import { fetchCourseDetail } from './data/thunks';
-import { useModel } from './generic/model-store';
 import NotFoundAlert from './generic/NotFoundAlert';
 import PermissionDeniedAlert from './generic/PermissionDeniedAlert';
 import { fetchOnlyStudioHomeData } from './studio-home/data/thunks';
 import { getCourseAppsApiStatus } from './pages-and-resources/data/selectors';
 import { RequestStatus } from './data/constants';
 import Loading from './generic/Loading';
+import { useCourseAuthoringContext } from './CourseAuthoringContext';
 
-const CourseAuthoringPage = ({ courseId, children }) => {
-  const dispatch = useDispatch();
+interface Props {
+  children?: React.ReactNode;
+}
 
-  useEffect(() => {
-    dispatch(fetchCourseDetail(courseId));
-  }, [courseId]);
+const CourseAuthoringPage = ({ children }: Props) => {
+  const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch(fetchOnlyStudioHomeData());
   }, []);
 
-  const courseDetail = useModel('courseDetails', courseId);
+  const { courseId, courseDetails, courseDetailStatus } = useCourseAuthoringContext();
 
-  const courseNumber = courseDetail ? courseDetail.number : null;
-  const courseOrg = courseDetail ? courseDetail.org : null;
-  const courseTitle = courseDetail ? courseDetail.name : courseId;
+  const courseNumber = courseDetails ? courseDetails.number : undefined;
```

Comment:
```suggestion
  const courseNumber = courseDetails?.number;
```
Optional; you can simplify this and the two lines below if you want.

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/2724/commits/8f0fc17c7e4d2064b658d6dc9ac7cbca9febd875